bgpd/bgp_route.c

```diff
@@ -3899,6 +3900,23 @@ void bgp_best_path_select_defer(struct bgp *bgp, afi_t afi, safi_t safi)
 		/* Send route processing complete message to RIB */
 		bgp_zebra_update(bgp, afi, safi,
 				 ZEBRA_CLIENT_ROUTE_UPDATE_COMPLETE);
+		bgp->gr_info[afi][safi].route_sync = true;
+
+		/* If this instance is all done, check for GR completion overall */
+		for (afi = AFI_IP; afi < AFI_MAX; afi++) {
+			for (safi = SAFI_UNICAST; safi <= SAFI_MPLS_VPN;
+			     safi++) {
+				if (bgp->gr_info[afi][safi].af_enabled &&
+				    !bgp->gr_info[afi][safi].route_sync) {
+					route_sync_pending = true;
+					break;
```

Comment:
I think it's not worth spending cycles on checking for other AFIs if we already know the desired state?

Reply:
This will come in handy when GR is supported for L2VPN EVPN AFI SAFI in the future.